src/vs/workbench/contrib/terminal/browser/xterm/markNavigationAddon.ts

```diff
@@ -50,23 +50,29 @@ export class CommandNavigationAddon extends Disposable implements ICommandTracke
 	}
 
 	private _refreshActiveCapability(store: ITerminalCapabilityStore) {
-		const activeCommandDetection = store.get(TerminalCapability.CommandDetection) || store.get(TerminalCapability.PartialCommandDetection);
-		if (activeCommandDetection !== this._commandDetection) {
-			this._commandDetection = activeCommandDetection;
+		const activeDetection = store.get(TerminalCapability.BufferMarkDetection) || store.get(TerminalCapability.CommandDetection) || store.get(TerminalCapability.PartialCommandDetection);
+		if (activeDetection !== this._detectionCapability) {
+			this._detectionCapability = activeDetection;
 		}
 	}
 
-	private _getCommandMarkers(): readonly IMarker[] {
-		if (!this._commandDetection) {
+	private _getMarkers(): readonly IMarker[] {
+		if (!this._detectionCapability) {
 			return [];
 		}
-		let commands: readonly IMarker[];
-		if (this._commandDetection.type === TerminalCapability.PartialCommandDetection) {
-			commands = this._commandDetection.commands;
-		} else {
-			commands = coalesce(this._commandDetection.commands.map(e => e.marker));
-		}
-		return commands;
+		let markers: readonly IMarker[] = [];
+		switch (this._detectionCapability.type) {
+			case TerminalCapability.PartialCommandDetection:
+				markers = this._detectionCapability.commands;
+				break;
+			case TerminalCapability.CommandDetection:
+				markers = coalesce(this._detectionCapability.commands.map(e => e.marker));
+				break;
+			case TerminalCapability.BufferMarkDetection:
+				markers = this._detectionCapability.markers();
+				break;
+		}
+		return markers;
```

Comment:
I don't think this is the behavior we're after, there are 2 modes for the command navigation addon to be in; partial or full command detection. You could argue what the best behavior is but imo for both modes, we want to navigate to all visible non-placeholder decorations. So cmd+up/down would typically navigate between commands, but if there are points or interest it will also navigate to them.